frontend/public/components/instantiate-template.tsx

```diff
@@ -206,7 +212,36 @@ class TemplateForm_ extends React.Component<
         },
       },
     };
-    return k8sCreate(TemplateInstanceModel, instance);
+    return k8sCreateResource({
+      model: TemplateInstanceModel,
+      data: instance,
+    });
+  }
+
+  updatesecretOwnerRef(
+    secret: K8sResourceKind,
+    templateInstance: K8sResourceKind,
+  ): Promise<K8sResourceKind> {
```

Comment:
I was hoping we could add the owner ref during creation, but looks like it's not possible to do that since we use generateName for both resources.

Reply:
@Lucifergene @rottencandy But the result of the create statements should return the created object with the "final" name?! Can you please take a look if we can use this?

Reply:
I think the other problem is that the secret needs to be created before the template instance, since the template instance requires a spec.secret field to be provided.